### DISCUSSION TITLE
dark: not overwriting bg-white

### DISCUSSION THREAD

**Discussion by gabr0236** [tailwindlabs/tailwindcss#10714] [Help] [answered]
Hello, I just started using tailwind today.

I'm trying to add dark mode to my project but encountering some problems.

I'm trying to add simple black/white darkmode like so `"bg-white text-black dark:bg-black dark:text-white"`. This is not working for me. 

But replacing the light mode colors works `"bg-red-400 text-green-400 dark:bg-black dark:text-white"` which is very strange.

Example: 
```
        <div class="bg-white text-black dark:bg-black dark:text-white">
            <p>TEST 1: "bg-white text-black dark:bg-black dark:text-white"</p>
        </div>
        <div class="bg-red-400 text-green-400 dark:bg-black dark:text-white">
            <p>TEST 2: "bg-red-400 text-green-400 dark:bg-black dark:text-white"</p>
        </div>
        <div>
            <p>IsDark: {{ isDark }}</p>
        </div>
```

Result of running:
- Darkmode off:
<img width="516" alt="Screenshot 2023-03-02 at 14 30 21" src="https://user-images.githubusercontent.com/61772220/222442393-b5ac1bbe-d023-4700-8c76-b6fcf55a6370.png">

- Darkmode on:
<img width="507" alt="Screenshot 2023-03-02 at 14 30 37" src="https://user-images.githubusercontent.com/61772220/222442450-01232530-9b87-41cf-b993-34564ef35672.png">

So the dark/light mode toggle is working, just not when the light color is either plain white or black.

On further inspection of the css, important color properties are added when using black/white:
<img width="417" alt="Screenshot 2023-03-02 at 14 32 57" src="https://user-images.githubusercontent.com/61772220/222443004-ee5440d7-a0d3-4c25-bd59-8afeb6b2a93b.png">

Using important selectors on the dark colors seems to work: `"bg-white text-black dark:!bg-black dark:!text-white"`.

But in the darkmode docs https://tailwindcss.com/docs/dark-mode no important selectors are necessary.

How to i fix this? 

Setup:
 - Using VueUse to toggle dark property on <html> tag.
 - tailwind.config
<img width="447" alt="Screenshot 2023-03-02 at 14 36 27" src="https://user-images.githubusercontent.com/61772220/222443789-482fa1f0-778b-404c-804a-70526563ccae.png">

**Comment by wongjn** [comment, marked as answer]
It seems like you have some rules which are interfering with Tailwind's rules:

```css
.text-black {
  color: #000000 !important;
}
```
```css
.bg-white {
  background-color: #FFFFFF !important;
  color: #000000 !important;
}
```
Try consilidating these rules, either by renaming the class names these rules use, or use a prefix with Tailwind to avoid this clash.

**Reply by gabr0236** [reply]
Thanks for your response! Yes this is the case. For some reason vuetify (node_modules/vuetify/lib/styles/main.css) has the following:

<img width="276" alt="Screenshot 2023-03-02 at 20 40 48" src="https://user-images.githubusercontent.com/61772220/222534841-29799d69-007b-4107-89a1-741652e18447.png">

Any idea how i can remove these default rules from vuetify (without having to remove vuetify completely). Will probably switch to another component lib soon. Maybe i should just overwrite using "bg-white text-black dark:!bg-black dark:!text-white" for now?

**Reply by wongjn** [reply]
> Any idea how i can remove these default rules from vuetify (without having to remove vuetify completely).

No I don't have any idea as I do not have familiarity with vuetify. Consider consulting their documentation.

> Maybe i should just overwrite using "bg-white text-black dark:!bg-black dark:!text-white" for now?

You could indeed.
